storyCite.lua

```diff
@@ -0,0 +1,248 @@
+				end
+				data["release year"] = data["release year"] .. " the years [[" .. releaseYear .. " (releases)|" .. releaseYear .."]] and [[" .. releaseEndYear .. " (releases)|" .. releaseEndYear .. "]]"
+			else
+				data["release year"] = "in the year [[" .. releaseYear .. " (releases)|" .. releaseYear .."]]"
+			end
+		else
+			data["release year"] = ""
+		end
+		
+		-- making outputText
+		local outputText = ""
+		if data["adapted from"] ~= "" then
+			outputText = outputText .. "Adapted from ''" .. data["adapted from"] .. "''"
+			if data["writer"] ~= "" then
+				outputText = outputText .. ", written by " .. data["writer"]
+				if data["publisher"] ~= "" then
+					outputText = outputText .. " and released by " .. data["publisher"]
+				else
+					outputText = outputText .. "and released "
+				end
+			elseif data["publisher"] ~= "" then
+				outputText = outputText .. " and released by " .. data["publisher"]
+			else
+				outputText = outputText .. "and released "
+			end
+		elseif data["writer"] ~= "" then
+			outputText = outputText .. "Written by " .. data["writer"]
+			if data["publisher"] ~= "" then
+				outputText = outputText .. " and released by " .. data["publisher"]
+			else
+				outputText = outputText .. "and released "
+			end
+		elseif data["publisher"] ~= "" then
+			outputText = outputText .. " Released by " .. data["publisher"]
+		else
+			outputText = outputText .. "Released "
+		end
+		if data["anthology"] ~= "" then
+				outputText = outputText .. " as part of ''" .. data["anthology"] .. "''"
+				if data["series"] ~= "" then
+					if (args["season number"] ~= "") or (args["season number"] ~= nil) then
+						outputText = outputText .. " from " .. data["series"]
+					else
+						outputText = outputText .. " from the series " .. data["series"]
+					end
+				end
+		elseif data["series"] ~= "" then
+			if (args["season number"] ~= "") or (args["season number"] ~= nil) then
+				outputText = outputText .. " as part of " .. data["series"]
+			else
+				outputText = outputText .. " as part of the series " .. data["series"]
+			end
+		end
+		if data["release year"] ~= "" then
+			outputText = outputText .. " " .. data["release year"]
+		end
+		if data["issues"] ~= "" then
+			outputText = outputText .. " throught " .. data["issues"]
+		end
+		--fallback in case of an error
+		if outputText == "" or outputText == nil then
+		 	outputText = "No data."
+		else
+			outputText = outputText .. "."
+		end
+		
+		--outputText = dump(data) --for testing
+		local name = frame.args[1]
+		local bin --used to store the output of the functions below
+		bin = util_vars.setVar(PREFIX .. name, outputText)
+		bin = cache.delete(PREFIX .. name)
+		bin = cache.set(PREFIX .. name, outputText)
+		bin = mw.smw.set("Story info", outputText) --documentation for this is down so I guessed the syntax. It doesn't seem to work.
+	end
+end
+
+function h.getInfo(story)
+	local info 
+	local bin --used to store the output of storing functions
```

Comment:
this isn't necessary; you don't have to set this in the outer scope unless you're going to use it in the outer scope. just say `local bin =` within the scope you need it later.

also, if you aren't going to use the output that a function returns, it's not required to store it into a variable. so you can just say `util_vars.setVar(PREFIX .. story, info)` without anything else on the line.